### DISCUSSION TITLE
`znap pull` command in a Zsh script?

### DISCUSSION THREAD

**Discussion by ghost** [marlonrichert/zsh-snap#34] [Q&A] [answered]
Is there a way to execute the `znap pull` command in a Zsh script?

Thank you.

**Comment by marlonrichert** [comment]
If you make sure that Znap has been sourced before you source the script, then yes, that should work. In Zsh, there’s no difference between calling commands from the command line, in your `.zshrc` file or from a script.

**Comment by ghost** [comment]
Oh, it's not a sourced script — it's a script file (where that command is) and I tried running it manually from the shell but I get an error that znap wasn't found.

I guess it won't work that way?... And there's no workaround?

**Comment by marlonrichert** [comment]
@andylima I have no idea what you’re talking about. Can you please give a concrete example?

**Comment by ghost** [comment]
These are the contents of the script file:
```
#!/usr/bin/env zsh

# upd zsh plugins using `znap`
# ie: upd all zsh plugin repos

echo "=== Updating Zsh plugins ==="
echo

znap pull
```

The name of that script file is `znu` and it's on a folder that's part of my `PATH`. When I type `znu` and press Return, I get this error:
```
command not found: znap
```

**Comment by marlonrichert** [comment, marked as answer]
@andylima I dug into it a bit and found out that what happens when you run your `znu` script as an executable, rather than `source`ing it, it gets executed in a subshell. The subshell inherits `export`ed parameters, but nothing else from the parent shell. So, for all practical purposes, when you run your script as such, it is as if Znap has never been initialized.

The easiest way around this is to convert your script to a function. So, instead of adding it to your `$path`, add it to your `$fpath` and `autoload` it. That way, it will run inside the same shell.

**Comment by ghost** [comment]
@marlonrichert I see. Thank you — I will use that workaround.
